app/github.server.ts

```diff
@@ -77,6 +77,52 @@ export async function getPullRequestInfo(
   return value.repository.pullRequest;
 }
 
+/**
+ * Accesses the "OSH Bot" GitHub app in order to make requests to the GitHub API
+ * without user credentials.
+ */
+async function getGitHubApp() {
+  if (!process.env.GITHUB_OSH_BOT_APP_ID) {
+    throw new Error("Missing GITHUB_OSH_BOT_APP_ID");
+  }
+  if (!process.env.GITHUB_OSH_BOT_PRIVATE_KEY_BASE_64) {
+    throw new Error("Missing GITHUB_OSH_BOT_PRIVATE_KEY_BASE_64");
+  }
+  if (!process.env.GITHUB_OSH_BOT_CLIENT_ID) {
+    throw new Error("Missing GITHUB_OSH_BOT_CLIENT_ID");
+  }
+  if (!process.env.GITHUB_OSH_BOT_CLIENT_SECRET) {
+    throw new Error("Missing GITHUB_OSH_BOT_CLIENT_SECRET");
+  }
+  if (!process.env.GITHUB_OSH_BOT_INSTALLATION_ID) {
+    throw new Error("Missing GITHUB_OSH_BOT_INSTALLATION_ID");
+  }
+
+  // The app id is a number
+  const parsedAppId = parseInt(process.env.GITHUB_OSH_BOT_APP_ID);
+
+  // The private key is encoded in base 64 so we convert it back to plain text
+  const parsedPrivateKey = Buffer.from(
+    process.env.GITHUB_OSH_BOT_PRIVATE_KEY_BASE_64,
+    "base64"
+  ).toString("utf-8");
+
+  const app = new App({
+    appId: parsedAppId,
+    privateKey: parsedPrivateKey,
+    oauth: {
+      clientId: process.env.GITHUB_OSH_BOT_CLIENT_ID,
+      clientSecret: process.env.GITHUB_OSH_BOT_CLIENT_SECRET,
```

Comment:
Just to double check. These and any other secrets are being stored… secretly and securely? How is that?

Reply:
They're stored in Vercel! Never exposed to contributors.